libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -86,10 +86,16 @@ struct OverrideGraph
 		int currentNode = static_cast<int>(numNodes++);
 		nodes[_function] = currentNode;
 		nodeInv[currentNode] = _function;
-		if (_function.overrides())
-			for (auto const& baseFunction: _function.baseFunctions())
+
+		bool addedEdge = false;
+		for (auto const& baseFunction: _function.baseFunctions())
+			if (_function.overrides() || baseFunction.contract().isInterface())
+			{
 				addEdge(currentNode, visit(baseFunction));
-		else
+				addedEdge = true;
+			}
+
+		if (!addedEdge)
```

Comment:
Can you perhaps explain this change as a comment?

Reply:
Basically, before we only considered whether the function itself has the `override` keyword, now we also alternatively check if the base function is in an interface. For that we have to do the check inside the loop, though.